### PR TITLE
tpch: clean up loader to be more idiomatic

### PR DESCRIPTION
**Pull request by rjnn**
Addresses unaddressed comment from #25.

---
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/loadgen/26)

### CODE REVIEW

File: tpch/load.go [hunk 1]
```diff
@@ -118,10 +108,21 @@ func insertTableFromFile(db *sql.DB, filename string, tableType table) error {
 			fields[i] = splits[i]
 		}
 
-		inserts[queuedInserts] = fmt.Sprintf(insertValues, fields...)
-		queuedInserts++
+		inserts = append(inserts, fmt.Sprintf(insertValues, fields...))
+		numTotalInserts++
+
+		if numTotalInserts%*insertsPerTransaction == 0 {
```
Comment (tamird) [inline tpch/load.go]: what is `%*`?

File: tpch/load.go [hunk 2]
```diff
@@ -54,8 +54,8 @@ func insertTableFromFile(db *sql.DB, filename string, tableType table) error {
 	}()
 
 	scanner := bufio.NewScanner(file)
-	var queuedInserts, numTotalInserts uint
-	inserts := make([]string, *insertsPerTransaction)
+	var numTotalInserts uint
```
Comment (tamird) [inline tpch/load.go:57]: this is `len(inserts)`, yes?

Comment (tamird) [reply tpch/load.go:57]: Ah, that slice is discarded sometimes.

File: tpch/load.go [hunk 3]
```diff
@@ -118,10 +108,21 @@ func insertTableFromFile(db *sql.DB, filename string, tableType table) error {
...
-		queuedInserts++
+		inserts = append(inserts, fmt.Sprintf(insertValues, fields...))
+		numTotalInserts++
+
+		if numTotalInserts%*insertsPerTransaction == 0 {
+			if err := doInserts(db, insertPreamble, inserts); err != nil {
+				return err
+			}
+			fmt.Printf("Inserts for table %2d:     %8d\n", tableType, numTotalInserts)
+			inserts = make([]string, 0, *insertsPerTransaction)
```
Comment (tamird) [inline tpch/load.go]: nit: `inserts = inserts[:0]`

File: tpch/load.go [hunk 4]
```diff
@@ -118,10 +108,21 @@ func insertTableFromFile(db *sql.DB, filename string, tableType table) error {
...
+		}
 	}
 
 	// Do any remaining inserts
-	return doInserts(db, insertPreamble, inserts)
+	if len(inserts) > 0 {
```
Comment (tamird) [inline tpch/load.go:124]: this check doesn't seem necessary.

**Comment by petermattis** [comment]
:lgtm:

---

Review status: 0 of 1 files reviewed at latest revision, 4 unresolved discussions.

---

*[tpch/load.go, line 114 at r1](https://reviewable.io:443/reviews/cockroachdb/loadgen/26#-KdgxflI6Xum5DeAjkWI-r1-114:-Kdgzp87mU3AFgKFkuj6:bcn83b7) ([raw file](https://github.com/cockroachdb/loadgen/blob/03d5c5af4e1dc597341c6aaef8f8a840d52cea68/tpch/load.go#L114)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

what is `%*`?
</blockquote></details>

Mod and deref. The `gofmt` spacing makes that confusing.

---

*[tpch/load.go, line 124 at r1](https://reviewable.io:443/reviews/cockroachdb/loadgen/26#-KdgxflI6Xum5DeAjkWI-r1-124:-Kdgzgzxhjp4fMFLBpHS:b1jr67s) ([raw file](https://github.com/cockroachdb/loadgen/blob/03d5c5af4e1dc597341c6aaef8f8a840d52cea68/tpch/load.go#L124)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

this check doesn't seem necessary.
</blockquote></details>

Why not? If it isn't done we'll send a malformed `INSERT`.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/loadgen/26#-:-KdgzvukbEIzjJGNjzsE:bnfp4nl)*

**Comment by rjnn** [comment]
Review status: 0 of 1 files reviewed at latest revision, 4 unresolved discussions.

---

*[tpch/load.go, line 114 at r1](https://reviewable.io:443/reviews/cockroachdb/loadgen/26#-KdgxflI6Xum5DeAjkWI-r1-114:-KdgziaCdpqH1jxyPTzE:bh5eq3k) ([raw file](https://github.com/cockroachdb/loadgen/blob/03d5c5af4e1dc597341c6aaef8f8a840d52cea68/tpch/load.go#L114)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

what is `%*`?
</blockquote></details>

added parens.

---

*[tpch/load.go, line 119 at r1](https://reviewable.io:443/reviews/cockroachdb/loadgen/26#-KdgxflI6Xum5DeAjkWI-r1-119:-KdgzlMSmadoRtsZG5CM:b-896fix) ([raw file](https://github.com/cockroachdb/loadgen/blob/03d5c5af4e1dc597341c6aaef8f8a840d52cea68/tpch/load.go#L119)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

nit: `inserts = inserts[:0]`
</blockquote></details>

Done.

---

*[tpch/load.go, line 124 at r1](https://reviewable.io:443/reviews/cockroachdb/loadgen/26#-KdgxflI6Xum5DeAjkWI-r1-124:-KdgznaDx5NCEH1Qay60:bfnznaf) ([raw file](https://github.com/cockroachdb/loadgen/blob/03d5c5af4e1dc597341c6aaef8f8a840d52cea68/tpch/load.go#L124)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

this check doesn't seem necessary.
</blockquote></details>

It is otherwise it does `insert into (...) VALUES` (with nothing after the `VALUES`) which is an error.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/loadgen/26)*

**Comment by tamird** [comment]
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/loadgen/26)*

**Pull request merged**